test/index.test.js

```diff
@@ -1958,32 +1958,187 @@ test('inline registration of component args - arrow function property lookup', a
+    `module.exports = EnhanceChildren`
+  ].join('\n')
+
+  const esx = [
+    `const esx = require('esx')()`,
+    `const { Children } = require("react")`,
+    'const EnhanceChildren = ({ children }) => Children.map(children, (child) => esx `<>${child}</>`)',
+    `module.exports = EnhanceChildren`
+  ].join('\n')
+
+  is(convert(src), esx);
+});
+
+test('Array as children', async ({ is }) => {
+  const src = [
+    `const React = require("react")`,
+    `const ArrayRender = () => [`,
+    `<li key="1">Item</li>,`,
+    `<li key="2">Item 2</li>,`,
+    `<li key="3">Item 3</li>,`,
+    `<li key="4">Item 4</li>,`,
+    `];`,
+    `module.exports = ArrayRender`,
+  ].join('\n')
+
+  const esx = [
+    `const esx = require('esx')();`,
+    `const React = require("react")`,
+    `const ArrayRender = () => [`,
+    'esx `<li key="1">Item</li>`,',
+    'esx `<li key="2">Item 2</li>`,',
+    'esx `<li key="3">Item 3</li>`,',
+    'esx `<li key="4">Item 4</li>`,',
+    `];`,
+    `module.exports = ArrayRender`,
+  ].join('\n');
+
+  is(convert(src), esx);
+});
+
+test('Nasty array as children', async ({ is }) => {
+  const src = [
+    `const React = require("react")`,
+    `const NastyArrayRender = () => [`,
+    `  <li key="1">Item</li>,`,
+    `  <li key="2">Item 2</li>,`,
+    `  <li key="3">Item 3</li>,`,
+    `  <li key="4">{[`,
+    `    <li key="n1">Nasty item1</li>,`,
+    `    <li key="n2">Nasty item2</li>,`,
+    `    <li key="n3">Nasty item3</li>,`,
+    `    <li key="n4">{[`,
+    `      <p>Foo</p>`,
+    `    ]}</li>,`,
+    `  ]}</li>`,
+    `];`,
+    `module.exports = NastyArrayRender`,
+  ].join('\n')
+
+  const esx = [
+    `const esx = require('esx')();`,
+    `const React = require("react")`,
+    `const NastyArrayRender = () => [`,
+    '  esx `<li key="1">Item</li>`,',
+    '  esx `<li key="2">Item 2</li>`,',
+    '  esx `<li key="3">Item 3</li>`,',
+    '  esx `<li key="4">${[',
+    '    esx `<li key="n1">Nasty item1</li>`,',
+    '    esx `<li key="n2">Nasty item2</li>`,',
+    '    esx `<li key="n3">Nasty item3</li>`,',
+    '    esx `<li key="n4">${[',
+    '      esx `<p>Foo</p>`',
+    '    ]}</li>`,',
+    '  ]}</li>`',
+    `];`,
+    `module.exports = NastyArrayRender`,
+  ].join('\n');
+
+  is(convert(src), esx);
+});
+
+
+test('Function as children', async ({ is }) => {
+  const src = [
+    `const React = require("react")`,
+    `const Repeater = (props) => <p>{props.children("foo")}</p>`,
+    'const App = () => <Repeater>{(str) => <h1>{str}</h1>}</Repeater>;',
+    `module.exports = App`,
+  ].join('\n');
+
+  const esx = [
+    `const esx = require('esx')();`,
+    'const React = require("react")',
+    'const Repeater = (props) => esx `<p>${props.children("foo")}</p>`',
+    `esx.register({ Repeater });`,
+    'const App = () => esx `<Repeater>${(str) => esx `<h1>${str}</h1>`}</Repeater>`;',
+    `module.exports = App`,
+  ].join('\n');
+
+  is(convert(src), esx);
+});
+
+test('fuckadoo', async ({ is }) => {
```

Comment:
we need to rename this hahah 

this was a case where it was failing, is it now passing? if it's passing for now let's call it 'complex case' 

if it is failing, let's whittle down the code to isolate the failure exactly, and then name the test accordingly

Reply:
@davidmarkclements got it, I renamed the test 

Yes, the tests are all passing even tho I would double check this precise case